spec/controllers/csv_files_controller_spec.rb

```diff
@@ -0,0 +1,265 @@
+require 'rails_helper'
+
+require 'support/controller_macros'
+require 'support/devise'
+require 'support/shared_examples_for_authentication'
+
+RSpec.describe CsvFilesController, type: :controller do
+  #############################################################################
+  ## generate_create_attributes
+  ## Generates attributes used in create methods. Since we cannot use the 
+  ## base CsvFile class, we use the WeamsCsvFile class.
+  #############################################################################
+  def generate_create_attributes(use_upload = true, factory = :weams_csv_file)
+
+    csv = attributes_for factory
+    csv[:type] = "WeamsCsvFile"
+
+    if !use_upload
+      csv[:upload] = nil 
+    else
+      csv[:upload].rewind
+    end
+
+    csv
+  end
+
+  #############################################################################
+  ## Define constant WeamsRawFile before any test.
+  #############################################################################
+  before(:all) do
+    class WeamsRawFile < CsvFile; end 
+  end
+
+  it_behaves_like "an authenticating controller", :index, "csv_files"
+
+  #############################################################################
+  ## index
+  #############################################################################
+  describe "GET index" do
+    login_user
+
+    before(:each) do
+      create :weams_csv_file
+      @csv = WeamsCsvFile.first
+
+      get :index
+    end
+
+    it "populates an array of csvs" do
+      expect(assigns(:csv_files)).to include(@csv)
```

Comment:
This line checks that the controller populates the variable :csv_files. The test preloads a single csv_file into the database, and the spec ensures that value is loaded into the array.